examples/creating-cert-chain-with-validation.cpp

```diff
@@ -0,0 +1,247 @@
+                // serial number of the root certificate
+                0,
+                rootCaSelfSignParams);
+}
+
+/*
+* Define CA Class for root CA based on root CA Certificate and keypair with random CA settings
+*/
+CertificateAuthority defineRootCa(const X509Certificate &rootCaCert,
+                                  const AsymmetricKeypair &rootCaKey)
+{
+    /* This basic constraints extension will be part of the signed certificate
+     * Set CA to true and path length to 0 (no more intermediate CA's)
+     * Remark: This will be the X509 content of intermediate CAs not of the root CA itself
+     */
+    BasicConstraintsExtension rootCertSignConstraints{true, 0};
+
+    // "Allow" certificate and certificate revocation list signing
+    const auto rootCaPrivateKeyUsage = KeyUsageExtension::Builder{}.keyCertSign().cRLSign().build();
+
+    auto rootCaSignParams = CertificateSigningParameters::Builder{}
+            .certificateValidity(Asn1Time::Seconds(90))
+            .digestType(digestType)
+            .addExtension(rootCertSignConstraints)
+            .addExtension(rootCaPrivateKeyUsage)
+            .build();
+
+    return CertificateAuthority(
+                rootCaSignParams,
+                // first serial number of the new certificates this CA signs
+                1,
+                rootCaCert,
+                rootCaKey);
+}
+
+/*
+* Sign (intermediate CA Cert) certificate with root CA based on given CSR
+*/
+X509Certificate signCertificateWithRootCa(const X509Certificate &signingCert,
+                                          const AsymmetricKeypair &signingKey,
+                                          const CertificateSigningRequest &csr)
+{
+    auto ca = defineRootCa(signingCert, signingKey);
+    try {
+        return ca.signCSR(csr);
+    } catch (const openssl::OpenSSLException &e) {
+        /* low level OpenSSL failure */
+        std::cerr << "Error reading CSR in PEM format. Please validate your CSR." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (const MoCOCrWException &e) {
+        /* Reason:
+         * - sanity checks of the certificate failed */
+        std::cerr << "Failed to create certificate." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+/*
+* Create signed intermediate CA Certificate
+*/
+X509Certificate createIntermediateCaCertificate(const AsymmetricKeypair &intermediateCaKey,
+                                                const AsymmetricKeypair &rootCaKey,
+                                                const X509Certificate &rootCert)
+{
+    auto intermediateCaCertDetails = DistinguishedName::Builder{}
+            .organizationalUnitName("ezcode")
+            .organizationName("eazz")
+            .countryName("DE")
+            .commonName("Common secondary certificate")
+            .build();
+
+    auto csr = CertificateSigningRequest(intermediateCaCertDetails,
```

Comment:
We could use this to explain here that things like the CA status (being a CA or not) are not part of the CSR but are rather configured by the issuer.